server/src/main/java/org/spine3/server/entity/storagefield/StorageFieldTraverser.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storagefield;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.Message;
+import org.spine3.annotations.SPI;
+import org.spine3.protobuf.AnyPacker;
+import org.spine3.server.entity.StorageFields;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A helper type for iterating over the {@link StorageFields} objects regarding the type
+ * of the entry.
+ *
+ * <p>This class has a set of methods {@code hitSmth}, where {@code Smth} is the type of the entry.
+ * To handle the types, implement those methods. They take a {@code String} param containing
+ * the name of the storage field and a {@linkplain Nullable nullable} value of the field.
+ *
+ * @author Dmytro Dashenkov.
+ */
+@SPI
+public abstract class StorageFieldTraverser {
```

Comment:
The purpose of this class is not clear, especially taking into account that it's marked `@SPI`. 

Let's discuss when the framework users ever need to define some custom strategy.